src/hotspot/share/runtime/thread.hpp

```diff
@@ -759,13 +759,14 @@ class Thread: public ThreadShadow {
 
  public:
   // Stack overflow support
-  address stack_base() const           { assert(_stack_base != NULL,"Sanity check"); return _stack_base; }
+  address stack_base() const           { return _stack_base; }
```

Comment:
Why did you remove the assertion? We want the assertion in general to ensure there are no improper uses of stack_base().

Reply:
We now reset NonJavaThread's stack base and size,  so _stack_base == NULL is possible, e.g. when generating hs_err report, as we should now see empty stack for "G1 Main Marker" thread in original bug report.

You do have a valid point for the assertion. I think following assertion is more precise:

assert(_stack_base != NULL || Thread::current() != NULL, "Sanity check");

What you think?

Thanks